### ISSUE TITLE
feat(models): MambaVesselNetAdapter — ModelAdapter ABC integration [T06]

### ISSUE THREAD

**Issue by petteriTeikari** [petteriTeikari/minivess-mlops#741]
## Summary

Wrap `MambaVesselNetBackbone` in a `MambaVesselNetAdapter(ModelAdapter)` that plugs into the platform's model-agnostic training loop. This is the integration layer that makes MambaVesselNet a first-class citizen alongside DynUNet, SegResNet, SAM3, and VesselFM — all without any changes to the training pipeline. Also registers the adapter in `model_builder.py` and verifies compatibility with `sliding_window_inference`, AMP policies, and checkpoint roundtrip.

## Context

- **Plan**: [`mambavesselnet-implementation-plan.xml`](docs/planning/mambavesselnet-implementation-plan.xml) — T06
- **Branch**: `feat/mambavesselnet`
- **Parent**: #737

## Acceptance Criteria

- [ ] `MambaVesselNetAdapter` uses `self.config` + `self.net` (base class requirements)
- [ ] `forward()` returns `SegmentationOutput` with softmax `prediction` + raw `logits`; shape `(B,C,H,W,D)`
- [ ] `get_eval_roi_size()` returns `(64, 64, 64)`
- [ ] `get_config()` returns `AdapterConfigInfo` with `family="mambavesselnet"`
- [ ] `export_onnx()` raises `NotImplementedError` with clear message (Mamba CUDA kernels not ONNX-traceable)
- [ ] `build_adapter(ModelConfig(family=MAMBAVESSELNET, ...))` dispatches correctly
- [ ] `_require_mamba()` raises `RuntimeError` with install instructions when mamba-ssm absent
- [ ] Lazy import: importing `model_builder` does NOT trigger `mamba-ssm` import
- [ ] `sliding_window_inference` test: `(1,1,128,128,128)` with `roi_size=(64,64,64)` → `(1,2,128,128,128)`, finite values
- [ ] AMP tests: BF16 forward no NaN; AMP-OFF validation no NaN; debug config has `mixed_precision_val=False`
- [ ] Checkpoint roundtrip: `adapter1` and reloaded `adapter2` outputs `allclose` (numerically identical)
- [ ] All CPU registry tests pass (staging tier); all GPU adapter tests pass

## Implementation Notes

**Required attribute names** (base class depends on these):
```python
self.config = config          # NOT self._config
self.net = MambaVesselNetBackbone(...)  # NOT any other name
```
`trainable_parameters()`, `load_checkpoint()`, `save_checkpoint()` are inherited for free via `self.net`.

Conservative AMP policy (same as SAM3): `mixed_precision=True` (training), `mixed_precision_val=False` (validation). Relax only if T06 AMP tests confirm no NaN with BF16 validation.

**Comment by petteriTeikari** [comment]
Closed: T06 complete (commit 233412b). MambaVesselNetAdapter implements ModelAdapter ABC; registered in model_builder; 14 CPU tests.
